### DISCUSSION TITLE
Output elastic energy

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#17330] [Q&A Modules: Phase field] [answered]
Hello MOOSE users,
In the [poly_grain_growth_2D_eldrforce.i ](https://github.com/idaholab/moose/blob/next/modules/combined/examples/phase_field-mechanics/poly_grain_growth_2D_eldrforce.i)input file, how to add a sub-module in the Auxkernel module to make the output cloud image file (*.e) show elastic energy?

And, the kernel responsible for elastic energy used in this input file is ACGrGrElasticDrivingForce, and the Residual is calculated as follows，
`    case Residual:
      return 0.5 *
             D_stress.doubleContraction(strain); // Compute the deformation energy driving force`

Very grateful for help!

Best Regards
Wei Peng

**Comment by jessecarterMOOSE** [comment, marked as answer]
Perhaps you could add an `ElasticEnergyMaterial` ([documentation](https://mooseframework.inl.gov/source/materials/ElasticEnergyMaterial.html)). From there you could output the material property directly to the exodus file by setting `outputs = exodus` in the `ElasticEnergyMaterial` block, or you could use an AuxKernel like `MaterialRealAux` ([documentation](https://mooseframework.inl.gov/source/auxkernels/MaterialRealAux.html)) to sample the elastic energy material.

**Reply by PengWei97** [reply]
Thank you for your answer, The problem has been solved. 
I added sub-blocks to the materals block, the code is as follows,
`[./elasticenergy]
  	type = GGElasticEnergyMaterial
        args = 'gr1 gr2'
        outputs = exodus
  [../]`
After that, F is displayed in the *.e file, which represents elastic energy.
Thanks again for your help.
Wei Peng
